Description: Add build flags for hardening and update docs.
 Use hardened flags to generate the binaries. Also fix the man page to
 include the description in the "NAME" tag. All these changes are to
 make lintian happy.

--- pacparser-1.3.5.orig/docs/html/doxygen.css
+++ pacparser-1.3.5/docs/html/doxygen.css
@@ -1,4 +1,4 @@
-/* The standard CSS for doxygen 1.8.9.1 */
+/* The standard CSS for doxygen 1.8.8 */
 
 body, table, div, p, dl {
 	font: 400 14px/22px Roboto,sans-serif;
@@ -227,7 +227,7 @@ span.lineno a:hover {
 	background-color: #C8C8C8;
 }
 
-div.ah, span.ah {
+div.ah {
 	background-color: black;
 	font-weight: bold;
 	color: #ffffff;
@@ -245,15 +245,6 @@ div.ah, span.ah {
 	background-image: -moz-linear-gradient(center top, #eee 0%, #444 40%, #000);
 }
 
-div.classindex ul {
-        list-style: none;
-        padding-left: 0;
-}
-
-div.classindex span.ai {
-        display: inline-block;
-}
-
 div.groupHeader {
 	margin-left: 16px;
 	margin-top: 12px;
@@ -782,7 +773,7 @@ div.directory {
     width: 24px;
     height: 18px;
     margin-bottom: 4px;
-    background-image:url('folderopen.png');
+    background-image:url('ftv2folderopen.png');
     background-position: 0px -4px;
     background-repeat: repeat-y;
     vertical-align:top;
@@ -793,7 +784,7 @@ div.directory {
     width: 24px;
     height: 18px;
     margin-bottom: 4px;
-    background-image:url('folderclosed.png');
+    background-image:url('ftv2folderclosed.png');
     background-position: 0px -4px;
     background-repeat: repeat-y;
     vertical-align:top;
@@ -804,7 +795,7 @@ div.directory {
     width: 24px;
     height: 18px;
     margin-bottom: 4px;
-    background-image:url('doc.png');
+    background-image:url('ftv2doc.png');
     background-position: 0px -4px;
     background-repeat: repeat-y;
     vertical-align:top;
--- pacparser-1.3.5.orig/docs/html/pacparser.html
+++ pacparser-1.3.5/docs/html/pacparser.html
@@ -3,7 +3,7 @@
 <head>
 <meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
 <meta http-equiv="X-UA-Compatible" content="IE=9"/>
-<meta name="generator" content="Doxygen 1.8.9.1"/>
+<meta name="generator" content="Doxygen 1.8.8"/>
 <title>Pacparser: pacparser</title>
 <link href="tabs.css" rel="stylesheet" type="text/css"/>
 <script type="text/javascript" src="jquery.js"></script>
@@ -26,7 +26,7 @@
 </table>
 </div>
 <!-- end header part -->
-<!-- Generated by Doxygen 1.8.9.1 -->
+<!-- Generated by Doxygen 1.8.8 -->
 </div><!-- top -->
 <div class="header">
   <div class="summary">
@@ -37,15 +37,15 @@
 </div><!--header-->
 <div class="contents">
 
-<p>API for pacparser library. A library to make web software PAC (proxy autoconfig) files intelligent. See <a href="http://code.google.com/p/pacparser">http://code.google.com/p/pacparser</a> for more information.  
+<p>API for pacparser library, a library to use proxy auto-config (PAC) files. See project homepage: <a href="http://github.com/pacparser/pacparser">http://github.com/pacparser/pacparser</a> for more information.  
 <a href="#details">More...</a></p>
 <table class="memberdecls">
 <tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
 Typedefs</h2></td></tr>
-<tr class="memitem:ga2b9fcff402478c85386354e439c82fc1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga2b9fcff402478c85386354e439c82fc1"></a>
-typedef int(*&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga2b9fcff402478c85386354e439c82fc1">pacparser_error_printer</a>) (const char *fmt, va_list argp)</td></tr>
-<tr class="memdesc:ga2b9fcff402478c85386354e439c82fc1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Type definition for pacparser_error_printer. <br /></td></tr>
-<tr class="separator:ga2b9fcff402478c85386354e439c82fc1"><td class="memSeparator" colspan="2">&#160;</td></tr>
+<tr class="memitem:ga943eb378f2ba61fb79a9f5f3ab848d13"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga943eb378f2ba61fb79a9f5f3ab848d13"></a>
+typedef int(*&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga943eb378f2ba61fb79a9f5f3ab848d13">pacparser_error_printer</a> )(const char *fmt, va_list argp)</td></tr>
+<tr class="memdesc:ga943eb378f2ba61fb79a9f5f3ab848d13"><td class="mdescLeft">&#160;</td><td class="mdescRight">Type definition for pacparser_error_printer. <br /></td></tr>
+<tr class="separator:ga943eb378f2ba61fb79a9f5f3ab848d13"><td class="memSeparator" colspan="2">&#160;</td></tr>
 </table><table class="memberdecls">
 <tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
 Functions</h2></td></tr>
@@ -73,7 +73,7 @@ Functions</h2></td></tr>
 <tr class="memitem:ga9ebafe5fd3ddef7613f569bdc58bb58b"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga9ebafe5fd3ddef7613f569bdc58bb58b">pacparser_setmyip</a> (const char *ip)</td></tr>
 <tr class="memdesc:ga9ebafe5fd3ddef7613f569bdc58bb58b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sets my IP address.  <a href="#ga9ebafe5fd3ddef7613f569bdc58bb58b">More...</a><br /></td></tr>
 <tr class="separator:ga9ebafe5fd3ddef7613f569bdc58bb58b"><td class="memSeparator" colspan="2">&#160;</td></tr>
-<tr class="memitem:ga24e658e648095306adb04ebdbcc9cf0f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga24e658e648095306adb04ebdbcc9cf0f">pacparser_set_error_printer</a> (<a class="el" href="#ga2b9fcff402478c85386354e439c82fc1">pacparser_error_printer</a> func)</td></tr>
+<tr class="memitem:ga24e658e648095306adb04ebdbcc9cf0f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga24e658e648095306adb04ebdbcc9cf0f">pacparser_set_error_printer</a> (<a class="el" href="#ga943eb378f2ba61fb79a9f5f3ab848d13">pacparser_error_printer</a> func)</td></tr>
 <tr class="memdesc:ga24e658e648095306adb04ebdbcc9cf0f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sets error printing function.  <a href="#ga24e658e648095306adb04ebdbcc9cf0f">More...</a><br /></td></tr>
 <tr class="separator:ga24e658e648095306adb04ebdbcc9cf0f"><td class="memSeparator" colspan="2">&#160;</td></tr>
 <tr class="memitem:ga5b6b6f53291893fd8e268805c98510c1"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="#ga5b6b6f53291893fd8e268805c98510c1">pacparser_enable_microsoft_extensions</a> (void)</td></tr>
@@ -84,7 +84,7 @@ Functions</h2></td></tr>
 <tr class="separator:ga11ed163d6a9440b1a0883fd474d3cc86"><td class="memSeparator" colspan="2">&#160;</td></tr>
 </table>
 <a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
-<p>API for pacparser library. A library to make web software PAC (proxy autoconfig) files intelligent. See <a href="http://code.google.com/p/pacparser">http://code.google.com/p/pacparser</a> for more information. </p>
+<p>API for pacparser library, a library to use proxy auto-config (PAC) files. See project homepage: <a href="http://github.com/pacparser/pacparser">http://github.com/pacparser/pacparser</a> for more information. </p>
 <dl class="section author"><dt>Author</dt><dd>Manu Garg <a href="#" onclick="location.href='mai'+'lto:'+'man'+'ug'+'arg'+'@g'+'mai'+'l.'+'com'; return false;">manug<span style="display: none;">.nosp@m.</span>arg@<span style="display: none;">.nosp@m.</span>gmail<span style="display: none;">.nosp@m.</span>.com</a> </dd></dl>
 <h2 class="groupheader">Function Documentation</h2>
 <a class="anchor" id="gaaa5640d8b9d78f6171d73b0a97c382c5"></a>
@@ -318,7 +318,7 @@ Functions</h2></td></tr>
         <tr>
           <td class="memname">void pacparser_set_error_printer </td>
           <td>(</td>
-          <td class="paramtype"><a class="el" href="#ga2b9fcff402478c85386354e439c82fc1">pacparser_error_printer</a>&#160;</td>
+          <td class="paramtype"><a class="el" href="#ga943eb378f2ba61fb79a9f5f3ab848d13">pacparser_error_printer</a>&#160;</td>
           <td class="paramname"><em>func</em></td><td>)</td>
           <td></td>
         </tr>
@@ -378,9 +378,9 @@ Functions</h2></td></tr>
 </div><!-- contents -->
 <!-- start footer part -->
 <hr class="footer"/><address class="footer"><small>
-Generated on Sat Apr 25 2015 18:10:02 for Pacparser by &#160;<a href="http://www.doxygen.org/index.html">
+Generated on Tue Sep 1 2015 18:56:47 for Pacparser by &#160;<a href="http://www.doxygen.org/index.html">
 Doxygen
-</a> 1.8.9.1
+</a> 1.8.8
 </small></address>
 </body>
 </html>
--- pacparser-1.3.5.orig/docs/man/man3/pacparser.3
+++ pacparser-1.3.5/docs/man/man3/pacparser.3
@@ -1,16 +1,16 @@
-.TH "pacparser" 3 "Sat Apr 25 2015" "Pacparser" \" -*- nroff -*-
+.TH "pacparser" 3 "Tue Sep 1 2015" "Pacparser" \" -*- nroff -*-
 .ad l
 .nh
 .SH NAME
-pacparser pacparser
+pacparser - Library to parse proxy auto-confg (PAC) files. 
 .PP
-API for pacparser library\&. A library to make web software PAC (proxy autoconfig) files intelligent\&. See http://code.google.com/p/pacparser for more information\&.  
+API for pacparser library, a library to use proxy auto-config (PAC) files\&. See project homepage: http://github.com/pacparser/pacparser for more information\&.  
 
 .SS "Typedefs"
 
 .in +1c
 .ti -1c
-.RI "typedef int(* \fBpacparser_error_printer\fP) (const char *fmt, va_list argp)"
+.RI "typedef int(* \fBpacparser_error_printer\fP )(const char *fmt, va_list argp)"
 .br
 .RI "\fIType definition for pacparser_error_printer\&. \fP"
 .in -1c
@@ -64,7 +64,7 @@ API for pacparser library\&. A library t
 .in -1c
 .SH "Detailed Description"
 .PP 
-API for pacparser library\&. A library to make web software PAC (proxy autoconfig) files intelligent\&. See http://code.google.com/p/pacparser for more information\&. 
+API for pacparser library, a library to use proxy auto-config (PAC) files\&. See project homepage: http://github.com/pacparser/pacparser for more information\&. 
 
 
 .PP
@@ -87,7 +87,7 @@ Initializes pac parser\&.
 .RE
 .PP
 Initializes JavaScript engine and does few basic initializations specific to pacparser\&. 
-.SS "int pacparser_parse_pac_file (const char * pacfile)"
+.SS "int pacparser_parse_pac_file (const char *pacfile)"
 
 .PP
 Parses the given PAC file\&. 
@@ -103,7 +103,7 @@ Parses the given PAC file\&.
 .RE
 .PP
 Reads the given PAC file and evaluates it in the JavaScript context created by pacparser_init\&. 
-.SS "int pacparser_parse_pac_string (const char * pacstring)"
+.SS "int pacparser_parse_pac_string (const char *pacstring)"
 
 .PP
 Parses the given PAC script string\&. 
@@ -119,7 +119,7 @@ Parses the given PAC script string\&.
 .RE
 .PP
 Evaulates the given PAC script string in the JavaScript context created by pacparser_init\&. 
-.SS "int pacparser_parse_pac (const char * pacfile)"
+.SS "int pacparser_parse_pac (const char *pacfile)"
 
 .PP
 Parses the gievn pac file\&. 
@@ -142,7 +142,7 @@ Use pacparser_parse_pac_file instead\&.
 .PP
 .PP
 Same as pacparser_parse_pac_file\&. Included only for backward compatibility\&. 
-.SS "char* pacparser_find_proxy (const char * url, const char * host)"
+.SS "char* pacparser_find_proxy (const char *url, const char *host)"
 
 .PP
 Finds proxy for the given URL and Host\&. 
@@ -160,7 +160,7 @@ proxy string on sucess and NULL on error
 .RE
 .PP
 Finds proxy for the given URL and Host\&. This function should be called only after pacparser engine has been initialized (using pacparser_init) and pac script has been parsed (using pacparser_parse_pac_file or pacparser_parse_pac_string)\&. 
-.SS "char* pacparser_just_find_proxy (const char * pacfile, const char * url, const char * host)"
+.SS "char* pacparser_just_find_proxy (const char *pacfile, const char *url, const char *host)"
 
 .PP
 Finds proxy for the given PAC file, URL and Host\&. 
@@ -184,7 +184,7 @@ This function is a wrapper around functi
 
 .PP
 Destroys JavaSctipt context\&. This function should be called once you're done with using pacparser engine\&. 
-.SS "void pacparser_setmyip (const char * ip)"
+.SS "void pacparser_setmyip (const char *ip)"
 
 .PP
 Sets my IP address\&. 
@@ -195,7 +195,7 @@ Sets my IP address\&.
 .RE
 .PP
 Sets my IP address to a custom value\&. This is the IP address returned by myIpAddress() javascript function\&. 
-.SS "void pacparser_set_error_printer (\fBpacparser_error_printer\fP func)"
+.SS "void pacparser_set_error_printer (\fBpacparser_error_printer\fPfunc)"
 
 .PP
 Sets error printing function\&. 
--- pacparser-1.3.5.orig/src/Makefile
+++ pacparser-1.3.5/src/Makefile
@@ -63,7 +63,6 @@ endif
 
 # Spidermonkey library.
 CFLAGS += -Ispidermonkey/js/src
-LDFLAGS += -lm
 
 LIBRARY_LINK = $(LIBRARY_NAME).$(SO_SUFFIX)
 PREFIX := $(DESTDIR)$(PREFIX)
@@ -91,13 +90,13 @@ pacparser.o: pacparser.c pac_utils.h pac
 	touch pymod/pacparser_o_buildstamp
 
 $(LIBRARY): pacparser.o libjs.a
-	$(MKSHLIB) $(LIB_OPTS) -o $(LIBRARY) pacparser.o libjs.a $(LDFLAGS)
+	$(MKSHLIB) $(CFLAGS) $(LDFLAGS) $(LIB_OPTS) -o $(LIBRARY) pacparser.o libjs.a -lm
 
 $(LIBRARY_LINK): $(LIBRARY)
 	ln -sf $(LIBRARY) $(LIBRARY_LINK)
 
 pactester: pactester.c pacparser.h $(LIBRARY_LINK)
-	$(CC) pactester.c -o pactester -lpacparser -L. -I.
+	$(CC) $(CFLAGS) $(LDFLAGS) pactester.c -o pactester -lpacparser -L. -I.
 
 testpactester: pactester
 	echo "Running tests for pactester."
--- pacparser-1.3.5.orig/src/pacparser.h
+++ pacparser-1.3.5/src/pacparser.h
@@ -29,9 +29,9 @@ extern "C" {
 
 /// @defgroup pacparser pacparser
 /// @{
-/// @brief API for pacparser library. A library to make web software PAC
-///	   (proxy autoconfig) files intelligent. See
-///	   http://code.google.com/p/pacparser for more information.
+/// @brief API for pacparser library, a library to use proxy auto-config (PAC)
+///       files. See project homepage: http://github.com/pacparser/pacparser
+///       for more information.
 /// @author Manu Garg <manugarg@gmail.com>
 
 /// @brief Initializes pac parser.
--- pacparser-1.3.5.orig/tools/generatedocs.sh
+++ pacparser-1.3.5/tools/generatedocs.sh
@@ -34,7 +34,7 @@ mv man/man3/* $docs_dir/man/man3/
 # Remove unnecessary and bad file deprecated.3
 rm -f $docs_dir/man/man3/deprecated.3
 # Fix man page.
-sed -i '' -e 's/pacparser \\\-/pacparser/g' $docs_dir/man/man3/*.3
+sed -i '' -e 's/pacparser \\\-/pacparser - Library to parse proxy auto-confg (PAC) files./g' $docs_dir/man/man3/*.3
 cd -
 
 echo $tmpdir
